### ISSUE TITLE
Validate that the same column is not specified in both indexes and uniqueIndexes

### ISSUE THREAD

**Issue by insidegui** [marcoarment/Blackbird#12]
TL;DR I had a hard time debugging a crash that was occurring while initializing the schema for my database. It looked like a crash related to a string enum not having an empty string case, but it was actually caused by having the same (unrelated) column listed in both `indexes` and `uniqueIndexes` for the affected model.

This is clearly a client mistake, however I think this could be validated in a way that makes it easier to spot, to prevent others from having to debug this on their own.

## The Problem

I had a model where I accidentally had `indexes` and `uniqueIndexes` repeat the same column, such as the example below:

```swift
struct Post: BlackbirdModel {
    static var primaryKey: [BlackbirdColumnKeyPath] = [ \.$guid ]

    static var indexes: [[BlackbirdColumnKeyPath]] = [
        [ \.$guid ],
        [ \.$title ],
    ]

    static var uniqueIndexes: [[BlackbirdColumnKeyPath]] = [
        [ \.$guid ],
    ]

    enum Format: String, BlackbirdStringEnum {
        case markdown
        case html
    }

    @BlackbirdColumn var guid: String
    @BlackbirdColumn var title: String
    @BlackbirdColumn var format: Format
}
```

When running the app, as soon as the model was touched (causing Blackbird to initialize the schema) I'd get a force unwrap crash here: https://github.com/marcoarment/Blackbird/blob/1f485e15c1a7b1f7bf12acd1251f267a287c84d6/Sources/Blackbird/BlackbirdCodable.swift#L132

In the example above, the crash would be caused by attempting to initialize a `BlackbirdStringEnum` with an empty string as its `rawValue`, but since there's no such case in that enum, this resulted in a crash.

I tracked it down to here:

https://github.com/marcoarment/Blackbird/blob/1f485e15c1a7b1f7bf12acd1251f267a287c84d6/Sources/Blackbird/BlackbirdSchema.swift#L315-L318

When a column is mistakenly repeated in `indexes` and `uniqueIndexes`, `currentIndexes != targetIndexes` will always be `true`, initiating the chain of events that lead to that crash.

I'm not familiar with the whole code base enough to send a PR right now, but wanted to document this here in case anyone else runs into this issue.

**Comment by marcoarment** [comment]
Fixed. Duplicate index definitions will now cause a fatalError with a helpful message.

Thanks!
